compiler/modules/front/token.bal

```diff
@@ -7,9 +7,10 @@ const DECIMAL_NUMBER = 1;
 const STRING_LITERAL = 2;
 const BOOLEAN_LITERAL = 3;
 const HEX_INT_LITERAL = 4;
+const DECIMAL_FP_NUMBER = 5;
 
 // Use string for DECIMAL_NUMBER so we don't get overflow on -int:MAX_VALUE
-type VariableLengthToken [IDENTIFIER, string]|[DECIMAL_NUMBER, string]|[STRING_LITERAL, string]|[HEX_INT_LITERAL, string];
+type VariableLengthToken [IDENTIFIER, string]|[DECIMAL_NUMBER, string]|[STRING_LITERAL, string]|[HEX_INT_LITERAL, string]|[DECIMAL_FP_NUMBER, float|string];
```

Comment:
On further reflection I think a cleaner way to represent the token is: `[DECIMAL_FP_NUMBER, string, FpTypeSuffix?]` where

```
type FpTypeSuffix "f";
```